src/components/AllComponents.js

```diff
@@ -0,0 +1,539 @@
+import React, { useEffect, useState } from "react";
+import SingleCard from "./SingleCard";
+import SuggestToWeb from "./SuggestToWeb";
+import { handleRarityChange } from "../util/helpers";
+
+import { useQuery, useLazyQuery } from "@apollo/client";
+import { QUERY_CHARACTERS, QUERY_ONECHARACTER, GET_USERDATA, GET_USERCHARACTERSBYID } from "../util/queries";
+import CardDetails from "./CardDetails";
+
+import Auth from "../util/auth";
+
+function AllComponents() {
+  const categorySelect = document.getElementById("categories");
+
+  const [search, setSearch] = useState("");
+  const [characters, setCharacters] = useState([]);
+  const [characterCategory, setCharacterCategory] = useState("");
+
+  const [rarityCategory, setRarityCategory] = useState("ALL");
+  const [filteredCharacters, setFilteredCharacters] = useState([]);
+  const [typeCharacters, setTypeCharacters] = useState("ALL");
+  const [cardDetails, setCardDetails] = useState({
+    id: 1331,
+    thumb: 1003310,
+    art: null,
+    name: "Buu (Kid)",
+    category: ['Resurrected Warriors', 'Majin Buu Saga', 'Transformation Boost', 'Artificial Life Forms', 'Majin Power', 'Planetary Destruction', 'Storied Figures', 'Legendary Existence', 'Sworn Enemies', 'Accelerated Battle', 'Worldwide Chaos', 'Battle of Fate'],
+    link_skill: ['Majin', 'Brutal Beatdown', 'More Than Meets the Eye', 'Big Bad Bosses', 'Infinite Regeneration', 'Fierce Battle', 'The Wall Standing Tall'],
+    type: "EPHY",
+    rarity: "UR",
+    ls_description: `Allies’ ATK increases (MAX +50%) based on HP left`,
+    sa_name: "Planet Burst",
+    sa_description: "Causes immense damage to enemy and lowers DEF  <Lowers enemy's DEF by 40% for 3 turns>  ",
+  });
+  const [suggestion, setSuggestion] = useState([]);
+  const [webOfTeam, setWebOfTeam] = useState([]);
+
+  const [urActive, setUrActive] = useState(false);
+  const [lrActive, setLrActive] = useState(false);
+
+  const [aglActive, setAglActive] = useState(false);
+  const [teqActive, setTeqActive] = useState(false);
+  const [intActive, setIntActive] = useState(false);
+  const [strActive, setStrActive] = useState(false);
+  const [phyActive, setPhyActive] = useState(false);
+  const [allTypeActive, setAllTypeActive] = useState(true);
+
+  
+  const { loading, data } = useQuery(QUERY_CHARACTERS);
+  const allCharacters = data?.characters || [];
+  
+  const [getUserData, { loading:loading3, data:data3 }] = useLazyQuery(GET_USERDATA);
```

Comment:
Similar to other components, recommend renaming the loading/data destructured items to more specific names